launcher/DesktopServices.h

```diff
@@ -3,31 +3,30 @@
 #include <QString>
 #include <QUrl>
 
+class QFileInfo;
+
 /**
  * This wraps around QDesktopServices and adds workarounds where needed
  * Use this instead of QDesktopServices!
  */
 namespace DesktopServices {
 /**
- * Open a file in whatever application is applicable
+ * Open a path in whatever application is applicable.
+ * @param ensurePathExists Make sure the path exists
  */
-bool openFile(const QString& path);
+bool openPath(const QFileInfo& path, bool ensurePathExists = false);
```

Comment:
```suggestion
bool openPath(const QFileInfo& path, bool makeFolder = false);
```
maybe rename the parameter so it's clear you shouldn't be using this when opening files